module/solutions/helper.js

```diff
@@ -1982,37 +1982,39 @@ module.exports = class SolutionsHelper {
         }
 
         let solutionDocument = await solutionsQueries.solutionDocuments(solutionQuery);
-
+        
         if (!solutionDocument[0]) {
           throw {
             message: messageConstants.apiResponses.SOLUTION_NOT_FOUND,
           };
         }
-
-        // let programQuery = {};
-
-        // let validateProgramId = gen.utils.isValidMongoId(programId);
-
-        // if (validateProgramId) {
-        //   programQuery["_id"] = programId;
-        // } else {
-        //   programQuery["externalId"] = programId;
-        // }
-
-        // let programDocument = await programsHelper.list(programQuery, [
-        //   "externalId",
-        //   "name",
-        //   "description",
-        //   "isAPrivateProgram",
-        // ]);
-
-        // if (!programDocument[0]) {
-        //   throw {
-        //     message: messageConstants.apiResponses.PROGRAM_NOT_FOUND,
-        //   };
-        // }
-
         let newSolutionDocument = _.cloneDeep(solutionDocument[0]);
+        let programQuery = {};
+        let programDocument;
+       if (programId !== ""){
+         let validateProgramId = gen.utils.isValidMongoId(programId);
```

Comment:
@praveenKDass this also can be optimized
const programQuery = gen.utils.isValidMongoId(programId) 
        ? { _id: programId } 
        : { externalId: programId };